### ISSUE TITLE
Add TypeScript Support for Compromise-Dates

### ISSUE THREAD

**Issue by dk4tz** [spencermountain/compromise#1067]
When using TypeScript and 'compromise-dates', the plugin methods do not appear on the base 'Three' type. This breaks compilation on TSC strict projects.

<img width="802" alt="image" src="https://github.com/spencermountain/compromise/assets/82852452/69288e4a-a024-4a54-94cb-3274aae72c1a">

**Comment by spencermountain** [comment]
agh, rats. Thanks @dk4tz I thought this was working.
To be honest, I'm still pretty crap at typescript, and may be already past my level of competence. You really seem to know your stuff - wanna take [a look](https://github.com/spencermountain/compromise/blob/master/plugins/dates/index.d.ts)? Any changes welcome, either advice or PR
cheers

**Comment by jose-elias-alvarez** [comment]
I am not a TypeScript expert either, but I think the issue is that the [return type annotation of `nlp`](https://github.com/spencermountain/compromise/blob/6239b8689df2c73ecfcd308726cad8bbe8a5dccd/types/three.d.ts#L5) uses a generic, which will always default to an empty object.

I'm not sure if this the idiomatic way to handle module augmentation, but this approach seems reasonable (and, importantly, works):

```typescript
// three.d.ts
declare function nlp(text: string, lexicon?: Lexicon): View & PluginMethods

declare module nlp {
  // other declarations
  export interface PluginMethods extends Object {}
}
```

Then, in the plugin's declarations:

```typescript
// index.d.ts (compromise-dates)
export interface DatesMethods {
  // methods go here
}

declare module 'compromise' {
  interface PluginMethods extends DatesMethods {}
}
```

This approach immediately augments the module when `compromise-dates` is pulled in.

One alternative could be to use the return type annotation of `nlp.plugin()` to infer the correct return type of subsequent calls to the object: 

```typescript
nlp.dates("today") // error
const nlpWithDates = nlp.plugin(datePlugin);
nlpWithDates.dates("today") // no error
```

This is technically safer, since it would only make TypeScript aware of plugin methods when a plugin is actually registered, but it also means that code in the wild would have to be rewritten to benefit. (It's also a little beyond my current TypeScript skills to implement, though I imagine I could figure it out.)

Either way, I'm happy to put in a PR if this sounds reasonable.

**Comment by spencermountain** [comment]
you do sound like a TypeScript expert! go for it!
thank you, very much.

**Comment by SitrucL** [comment]
Hi was there ever a PR made for this? I'm still seeing similar behaviour to the OP

<img width="1086" alt="Image" src="https://github.com/user-attachments/assets/efff21d6-b546-4e13-9f54-c416a21e93af" />

**Comment by spencermountain** [comment]
Yeah, I could really use some help with this stuff, I have no idea how to do this sort of stuff in typescript.

The good news is that `Three` doesn't have a dates method. The bad news is that we modify the prototype when we install the dates plug-in, and I don't know how to accomplish this properly in typescript config. I'm sure there's a way to do it, but it also seems to change with the seasons.

A solution could be to use the same format in the date plugin, as recently changed [here](https://github.com/spencermountain/compromise/pull/1165).
![Image](https://github.com/user-attachments/assets/f26e7e6a-d228-439f-bff1-92d9aa66ede8)
Honestly, i have no idea.

If anyone can lend hand, I'll merge anything! I don't think it's wise for me to go in and change things. haha
Cheers
